src/tiled/newtemplatedialog.cpp

```diff
@@ -73,9 +82,50 @@ void NewTemplateDialog::updateTemplateName()
 void NewTemplateDialog::updateOkButton()
 {
     QPushButton *okButton = mUi->buttonBox->button(QDialogButtonBox::Ok);
-    QString templateName = mUi->templateName->text();
 
+    bool noTemplateName = mUi->templateName->text().isEmpty();
     int index = mUi->groupsComboBox->currentIndex();
 
-    okButton->setEnabled(!templateName.isEmpty() && index != -1);
+    okButton->setDisabled(noTemplateName || index==-1);
+}
+
+/**
+ * This code is duplicated from the templates dock.
+ */
+void NewTemplateDialog::newTemplateGroup()
+{
+    QString filter = TtxTemplateGroupFormat::instance()->nameFilter();
+
+    Preferences *prefs = Preferences::instance();
+    QString suggestedFileName = prefs->lastPath(Preferences::TemplateDocumentsFile);
+    suggestedFileName += QLatin1String("/untitled.ttx");
+
+    QString fileName = QFileDialog::getSaveFileName(this, tr("Save File"),
+                                                    suggestedFileName,
+                                                    filter);
+
+    if (fileName.isEmpty())
+        return;
+
+    auto templateGroup = new TemplateGroup();
+    auto name = QFileInfo(fileName).baseName();
+    templateGroup->setName(name);
+    templateGroup->setFileName(fileName);
+    QScopedPointer<TemplateGroupDocument>
+        templateGroupDocument(new TemplateGroupDocument(templateGroup));
+
+    QString error;
+    if (!templateGroupDocument->save(fileName, &error)) {
+        QMessageBox::critical(this, tr("Error Creating Template Group"), error);
+        return;
+    }
+
+    auto model = ObjectTemplateModel::instance();
+    model->addNewDocument(templateGroupDocument.take());
+
+    prefs->setLastPath(Preferences::TemplateDocumentsFile,
+                       QFileInfo(fileName).path());
+
+    mUi->groupsComboBox->addItem(name);
+    mUi->groupsComboBox->setCurrentIndex(mUi->groupsComboBox->count() - 1);
```

Comment:
After this it should call `updateOkButton` (or connect the index change with this function).